### ISSUE TITLE
visual map not work in quickfix or floaterm window

### ISSUE THREAD

**Issue by aceforeverd** [tversteeg/registers.nvim#51]
Hi, here is my experience with visual map in quickfix or floaterm window that I feel not right. 

## vimrc

```vim
if empty(glob('~/.config/nvim/autoload/plug.vim'))
    silent execute '!curl -fLo '.s:home.'/autoload/plug.vim --create-dirs  https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim'
    autocmd VimEnter * PlugInstall --sync | source $MYVIMRC
endif

call plug#begin('~/.config/nvim/bundle/pkgs/')

Plug 'tversteeg/registers.nvim'

call plug#end()

filetype plugin on
filetype indent on
```

## reproduce

1. vim open a file
2. `:grep something` in a file
2. `:copen`
3. visually select a line then `"+`
4. after hitting `+` the cursor jump to editing file from quickfix window.

some problem happen from a floaterm window, it go the insert mode instead yank the text

## env

I test with nvim-0.7.0 but it should also reproduce with nvim 0.6

**Comment by tversteeg** [comment]
I've added a change that won't open the registers window when the buffer is not modifiable. This should fix this issue, can you verify that that's the case?

**Comment by aceforeverd** [comment]
thanks for the fix and I test with the same vimrc, but It seems not work

**Comment by tversteeg** [comment]
Do you get the exact same problem? Where the cursor jumps back? Is this only in the `:copen` window?

**Comment by aceforeverd** [comment]
> Do you get the exact same problem? Where the cursor jumps back? Is this only in the `:copen` window?

slightly different. the problem exists still in `:copen` or floterm window, but this time I press `"` after visually select a few lines, it quit and switch back to normal mode immediately

**Comment by aceforeverd** [comment]
Hi I just investigated a little, and here is I found
1. floaterm and quickfix may have different reason
2. for floaterm, my steps is `<C-\><C-n>` -> `Vjjjj` -> `"` -> `+y`,  the reason is it go back to terminal mode instead of terminal normal mode, just as `vim.api.nvim_win_close` will go back to terminal mode directly, that's why I finally got `gv"+y` in the terminal prompt.  Cuz the problem does not exist in native nvim terminal, I don't know if it's floterm's problem.
3. for quickfix, I don't have a collusion yet.  But I met 2 problems over ther
   - one is during `Vjjjj` -> `"`, after press `"`, the cursor jump to edit buffer and popup appear there.  It seems open window with `height = 25` in my environment, when I manually change the `height` option to 10, the popup window appear on top of quick window.
   - after change `height` to 10, the cursor still jump above, but just after press the specific register key, like `Vjjj` -> `"" -> `*`
